packages/cactus-plugin-ledger-connector-iroha/src/test/typescript/integration/run-transaction-endpoint-v1.test.ts

```diff
@@ -0,0 +1,783 @@
+    t.ok(res);
+    t.ok(res.data);
+    t.equal(res.status, 200);
+    t.deepEqual(res.data.transactionReceipt, [
+      {
+        assetId: "coolcoin#test",
+        accountId: "admin@test",
+        balance: "35.250",
+      },
+    ]);
+  }
+
+  {
+    const req = {
+      commandName: "getSignatories",
+      params: ["admin@test"],
+    };
+    const res = await apiClient.runTransactionV1(req as RunTransactionRequest);
+    t.ok(res);
+    t.ok(res.data);
+    t.equal(res.status, 200);
+    t.deepEquals(res.data.transactionReceipt, [
+      "313a07e6384776ed95447710d15e59148473ccfc052a681317a72a69f2a49910",
+    ]);
+  }
+
+  {
+    const req = {
+      commandName: "addSignatory",
+      params: [
+        "admin@test",
+        "0000000000000000000000000000000000000000000000000000000000000001",
+      ],
+    };
+    const res = await apiClient.runTransactionV1(req as RunTransactionRequest);
+    t.ok(res);
+    t.ok(res.data);
+    t.equal(res.status, 200);
+    t.equal(res.data.transactionReceipt.status, "COMMITTED");
+  }
+
+  {
+    const req = {
+      commandName: "getSignatories",
+      params: ["admin@test"],
+    };
+    const res = await apiClient.runTransactionV1(req as RunTransactionRequest);
+    t.ok(res);
+    t.ok(res.data);
+    t.equal(res.status, 200);
+    t.deepEqual(res.data.transactionReceipt, [
+      "0000000000000000000000000000000000000000000000000000000000000001",
+      "313a07e6384776ed95447710d15e59148473ccfc052a681317a72a69f2a49910",
+    ]);
+  }
+
+  {
+    const req = {
+      commandName: "removeSignatory",
+      params: [
+        "admin@test",
+        "0000000000000000000000000000000000000000000000000000000000000001",
+      ],
+    };
+    const res = await apiClient.runTransactionV1(req as RunTransactionRequest);
+    t.ok(res);
+    t.ok(res.data);
+    t.equal(res.status, 200);
+    t.equal(res.data.transactionReceipt.status, "COMMITTED");
+  }
+
+  {
+    const req = {
+      commandName: "getSignatories",
+      params: ["admin@test"],
+    };
+    const res = await apiClient.runTransactionV1(req as RunTransactionRequest);
+    t.ok(res);
+    t.ok(res.data);
+    t.equal(res.status, 200);
+    t.deepEqual(res.data.transactionReceipt, [
+      "313a07e6384776ed95447710d15e59148473ccfc052a681317a72a69f2a49910",
+    ]);
+  }
+
+  {
+    const req = {
+      commandName: "getRoles",
+      params: [],
+    };
+    const res = await apiClient.runTransactionV1(req as RunTransactionRequest);
+    t.ok(res);
+    t.ok(res.data);
+    t.equal(res.status, 200);
+    t.deepEqual(res.data.transactionReceipt, [
+      "cactus_test",
+      "cactus_test_full",
+      "admin",
+      "user",
+      "money_creator",
+    ]);
+  }
+
+  {
+    const req = {
+      commandName: "getRolePermissions",
+      params: ["user"],
+    };
+    const res = await apiClient.runTransactionV1(req as RunTransactionRequest);
+    t.ok(res);
+    t.ok(res.data);
+    t.equal(res.status, 200);
+    const testArr = [
```

Comment:
@hxlaf A comment here would be good to explain what is this array with numbers. I have a few ideas on where these numbers are coming from, but I'm not a 100% sure TBH (maybe it's an Iroha thing)

Reply:
> @hxlaf A comment here would be good to explain what is this array with numbers. I have a few ideas on where these numbers are coming from, but I'm not a 100% sure TBH (maybe it's an Iroha thing)

Sure! I can do comments about it. Indeed, this is a Iroha thing. Please see the screenshot below. They have a role mapping (role name <-> index number)
![image](https://user-images.githubusercontent.com/10954157/127261144-563bef36-00ac-4b55-b7ab-7b4f296c41a4.png)
More specifically, this file: [.](https://github.com/hyperledger/iroha-javascript/blob/master/src/proto/primitive_pb.d.ts#L193-L247) 
We can talk about it this week, maybe we can do some translation to make it more readable?

Reply:
Comments added!
![image](https://user-images.githubusercontent.com/10954157/127266820-4543cc6b-25b0-4895-bf13-456753228f15.png)